src/main/java/com/minecolonies/coremod/client/gui/townhall/WindowPermissionsPage.java

```diff
@@ -320,6 +335,23 @@ public void updateElement(final int i, final Pane pane)
                 pane.findPaneOfTypeByID("rankId", Text.class).setText(Component.literal(Integer.toString(rank.getId())));
             }
         });
+
+        DropDownList dropdown = findPaneOfTypeByID(TOWNHALL_RANK_TYPE_PICKER, DropDownList.class);
+        dropdown.setDataProvider(new DropDownList.DataProvider()
+        {
+            @Override
+            public int getElementCount()
+            {
+                return rankTypes.size();
+            }
+
+            @Override
+            public String getLabel(final int i)
```

Comment:
Use getLabelNew if returning components, need to deprecate the old one (or remove it finally after few yrs xd)

Reply:
getLabel is required

Reply:
Tru, but not called if you override component variant